Makefile

```diff
@@ -1,6 +1,10 @@
 .PHONY: all
-
 VERSION := 4.2.0
+VERSION_FULL ?= $(shell test -e .git && git describe --tags --long)
```

Comment:
@mslacken is it possible that "git describe" should only have the "--first-parent" option set?  see: https://git-scm.com/docs/git-describe/2.33.0